packages/keystone/lib/Keystone/index.js

```diff
@@ -228,20 +228,50 @@ module.exports = class Keystone {
     };
   }
 
+  createContext({ schemaName = 'public', authentication = {}, skipAccessControl = false }) {
+    const context = {
+      schemaName,
+      authedItem: authentication.item,
+      authedListKey: authentication.listKey,
+      ...this._getAccessControlContext({ schemaName, authentication, skipAccessControl }),
+      totalResults: 0,
+      maxTotalResults: this.queryLimits.maxTotalResults,
+    };
+    const orig = { schemaName, authentication, skipAccessControl };
+    context.createContext = ({
+      schemaName = orig.schemaName,
+      authentication = orig.authentication,
+      skipAccessControl = orig.skipAccessControl,
+    }) => this.createContext({ schemaName, authentication, skipAccessControl });
+    context.executeGraphQL = ({ context = context, query, variables }) =>
+      this.executeGraphQL({ context, query, variables });
+    return context;
+  }
+
+  executeGraphQL({ context, query, variables }) {
+    if (!context) {
+      context = this.createContext();
```

Comment:
I think you want  ```context = this.createContext({});```

Reply:
Ooh, good catch, I think you're right!

Reply:
@timleslie starting a new graphql project and running ` npx keystone create-tables` results in 
```
TypeError: Cannot read property 'schemaName' of undefined
    at Keystone.createContext (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/lib/Keystone/index.js:231:19)
    at Keystone.executeGraphQL (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/lib/Keystone/index.js:255:22)
    at Object.module.exports [as onConnect] (/Users/levyklots/my-app/initial-data.js:10:22)
    at Keystone.connect (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/lib/Keystone/index.js:540:33)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async createTables (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/bin/commands/create-tables.js:11:3)
```

Reply:
Yep, that'll do it, let me ship a fix for this now. Thanks for the bug report 🙏

Reply:
@lklots I've published `@keystonejs/keystone@11.1.3` which fixes this issue. Thanks again for the bug report!

Reply:
@timleslie there is another error if you run `npx keystone create-tables` in a brand new adminui project below. I'll start debugging and let you know but wanted you to be aware


```
✖  
Error: No executable schema named 'public' is available. Have you setup '@keystonejs/app-graphql'?
    at Keystone.executeGraphQL (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/lib/Keystone/index.js:259:13)
    at Object.module.exports [as onConnect] (/Users/levyklots/my-app/initial-data.js:10:22)
    at Keystone.connect (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/lib/Keystone/index.js:540:33)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async createTables (/Users/levyklots/my-app/node_modules/@keystonejs/keystone/bin/commands/create-tables.js:11:3)
```

Reply:
😓 This is also definitely a bug, but it's a little more confusing. In particular, I'm wondering how the code that we had *before* the change to `executeGraphQL` ever worked! We shouldn't be trying to initialise data via server side graphQL operations in the `onConnect` phase, because at this point we haven't actually set up our graphQL schemas yet! That happens at a later phase of the system set up.

I think that perhaps that it *didn't* work in the past, but it was failing silently, rather than throwing an error, which is even worse!

Let me dig into this and try and work out what's going on and how to fix it.

Reply:
@timleslie that's for sure what's happening since the errors are being gobbled up when running `yarn dev` since it is not reporting failures on `npx keystone create-tables`

```
yarn dev
yarn run v1.22.4
$ cross-env NODE_ENV=development DISABLE_LOGGING=true keystone dev
ℹ Command: keystone dev
✔ Validated project entry file ./index.js
✔ Keystone server listening on port 3000
No cookieSecret value was provided. Please generate a secure value and add it to your app. Until this is done, a random cookieSecret will be generated each time Keystone is started. This will cause sessions to be reset between restarts. See [https://www.keystonejs.com/keystonejs/keystone/#config] for details.
✔ Initialised Keystone instance
✔ Connected to database
⠋ Verifying tables

Database is empty. Initialise tables with: 'npx keystone create-tables'
See: 🔗 https://www.keystonejs.com/quick-start/#installing-keystone


```

Reply:
I've been able to reproduce the error you're getting, which is a good start.

Reply:
I think the bug is here:

```
  onConnect: process.env.CREATE_TABLES !== 'true' && initialiseData,
```

When we run `create-tables` we connect directly to the database, without ever bothering to setup graphQL servers (since all we're doing is executing some database commands).

As such, our start app signals to only run `initialiseData` when we're *not* running `create-tables`.

If you run:

```
yarn create-tables
```

or

```
npm run create-tables
```

then it will run the script:

```
    "create-tables": "CREATE_TABLES=true keystone create-tables"
```
from `package.json`, which sets up the env var correctly, and you should be able to successfully create your tables.

The *real* culprit here is the bad instructions on the quick-start guide page which say:

```
yarn keystone create-tables
# or
npx keystone create-tables
```

This will work fine for projects which don't have anything in their `onConnect` function, but happen to be bad advice in the case of the `starter` project, which requires that env-var.

I will look into resolving this, either by updating the docs on the quick-start page, or finding a better mechanism that plays nicely with the existing instructions.

Reply:
@timleslie you're awesome thank you!

Reply:
Problem solved in https://github.com/keystonejs/keystone/pull/3196